src/app/component/CartItem/CartItem.container.js

```diff
@@ -133,8 +142,22 @@ export class CartItemContainer extends PureComponent {
      */
     handleRemoveItem() {
         this.setState({ isLoading: true }, () => {
-            const { removeProduct, item: { item_id } } = this.props;
-            this.hideLoaderAfterPromise(removeProduct(item_id));
+            const {
+                removeProduct,
+                updateCrosssellProducts,
+                updateCrossselsOnRemove,
+                item: { item_id }
+            } = this.props;
+
+            this.hideLoaderAfterPromise((async () => {
+                const result = await removeProduct(item_id);
+
+                if (result && updateCrossselsOnRemove) {
+                    await updateCrosssellProducts(result.items);
+                }
+
+                return result;
+            })());
```

Comment:
It is better to have this callback declared on the class, in the current way it is hardly extendable by e.g. plugins.

Reply:
Done